src/libraries/System.Private.CoreLib/src/System/Numerics/Decimal128.cs

```diff
@@ -0,0 +1,198 @@
+
+        static int IDecimalIeee754ConstructorInfo<Decimal128, Int128, UInt128>.CountDigits(Int128 number) => FormattingHelpers.CountDigits((UInt128)number);
+
+        static Int128 IDecimalIeee754ConstructorInfo<Decimal128, Int128, UInt128>.Power10(int exponent) => Int128Powers10[exponent];
+
+        static Int128 IDecimalIeee754ConstructorInfo<Decimal128, Int128, UInt128>.MaxSignificand => MaxSignificand;
+
+        static int IDecimalIeee754ConstructorInfo<Decimal128, Int128, UInt128>.MaxDecimalExponent => MaxDecimalExponent;
+
+        static int IDecimalIeee754ConstructorInfo<Decimal128, Int128, UInt128>.MinDecimalExponent => MinDecimalExponent;
+
+        static int IDecimalIeee754ConstructorInfo<Decimal128, Int128, UInt128>.NumberDigitsPrecision => NumberDigitsPrecision;
+
+        static int IDecimalIeee754ConstructorInfo<Decimal128, Int128, UInt128>.Bias => Bias;
+
+        static int IDecimalIeee754ConstructorInfo<Decimal128, Int128, UInt128>.MostSignificantBitNumberOfSignificand => 113;
+
+        static int IDecimalIeee754ConstructorInfo<Decimal128, Int128, UInt128>.NumberBitsEncoding => 128;
+
+        static int IDecimalIeee754ConstructorInfo<Decimal128, Int128, UInt128>.NumberBitsCombinationField => 17;
+
+        static int IDecimalIeee754ConstructorInfo<Decimal128, Int128, UInt128>.NumberBitsExponent => NumberBitsExponent;
+
+        static UInt128 IDecimalIeee754ConstructorInfo<Decimal128, Int128, UInt128>.PositiveInfinityBits => PositiveInfinityValue;
+
+        static UInt128 IDecimalIeee754ConstructorInfo<Decimal128, Int128, UInt128>.NegativeInfinityBits => NegativeInfinityValue;
+
+        static UInt128 IDecimalIeee754ConstructorInfo<Decimal128, Int128, UInt128>.Zero => ZeroValue;
+
+        static int IDecimalIeee754UnpackInfo<Decimal128, Int128, UInt128>.ConvertToExponent(UInt128 value) => (int)value;
+
+        static Int128 IDecimalIeee754UnpackInfo<Decimal128, Int128, UInt128>.ConvertToSignificand(UInt128 value) => (Int128)value;
+
+        static Int128 IDecimalIeee754UnpackInfo<Decimal128, Int128, UInt128>.Power10(int exponent) => Int128Powers10[exponent];
+
+        static UInt128 IDecimalIeee754UnpackInfo<Decimal128, Int128, UInt128>.SignMask => new UInt128(0x8000_0000_0000_0000, 0);
+
+        static int IDecimalIeee754UnpackInfo<Decimal128, Int128, UInt128>.NumberBitsEncoding => 128;
+
+        static int IDecimalIeee754UnpackInfo<Decimal128, Int128, UInt128>.NumberBitsExponent => NumberBitsExponent;
+
+        static int IDecimalIeee754UnpackInfo<Decimal128, Int128, UInt128>.Bias => Bias;
+
+        static Int128 IDecimalIeee754UnpackInfo<Decimal128, Int128, UInt128>.TwoPowerMostSignificantBitNumberOfSignificand => new Int128(0x0002_0000_0000_0000, 0);
+
+        static int IDecimalIeee754UnpackInfo<Decimal128, Int128, UInt128>.NumberDigitsPrecision => NumberDigitsPrecision;
+
+
+        private static Int128[] Int128Powers10 =>
```

Comment:
If Int128 is not optimized, this can be optimally stored as a ReadOnlySpan of Int32/Int64 like Number.BigInteger.Pow10BigNumTable